zstd.go

```diff
@@ -60,8 +60,8 @@ func cCompressBound(srcSize int) int {
 // decompressSizeHint tries to give a hint on how much of the output buffer size we should have
 // based on zstd frame descriptors. To prevent DOS from maliciously-created payloads, limit the size
 func decompressSizeHint(src []byte) int {
-	// 1 MB or 10x input size
-	upperBound := 10 * len(src)
+	// 1 MB or 50x input size
+	upperBound := 50 * len(src)
```

Comment:
That's a huge bump no? It means we'll allocate 5x more memory than before in all these cases.

Reply:
We will in the case where we cannot read the size from the zstd headers (or that the size we read is bigger than 10x the compressed size).

However imo it's better to do that and allocate once rather than first allocate 10*size, realise it's not enough and reallocate 50*size